lib/pbench/server/database/models/datasets.py

```diff
@@ -186,17 +184,36 @@ def __str__(self) -> str:
         return f"No metadata {self.key} for {self.dataset}"
 
 
+class MetadataBadStructure(MetadataError):
+    """
+    A call to `getvalue` or `setvalue` found a level in the JSON document where
+    the caller's key expected a nested JSON object but the type at that level
+    is something else. For example, when `user.contact.email` finds that
+    `user.contact` is a string, it's impossible to look up the `email` field.
+
+    The error text will identify the key path and the expected key element that
+    is missing.
+    """
+
+    def __init__(self, dataset: "Dataset", path: str, element: str):
+        super().__init__(dataset, path)
+        self.element = element
+
+    def __str__(self) -> str:
+        return f"Key {self.element!r} value for {self.key!r} in {self.dataset} is not a JSON object"
```

Comment:
I find the reference to `key` at line 203 jarring, because there is no obvious antecedent for it in the surrounding code.  The problem is that the value of `path` gets stored as `key`, but that's not obvious unless one looks at the superclass implementation.  I can guess why this is the way it is, but that doesn't actually solve the problem.  So, if you modify this PR for some other reason, please consider renaming `path` to `key`.

Reply:
I successfully ignored this concern on my initial reviews, but, when I saw it again and still was not at peace with it, I felt impelled to mention it.  However, there is no strong push, here.